exchanges/oex/oex.go

```diff
@@ -0,0 +1,501 @@
+	}
+
+	if resp.Error != oexNoError {
+		return resp, errors.New(resp.ErrCapture.Msg)
+	}
+
+	return resp, nil
+}
+
+// GetUserInfo gets account information given the API key and Secret
+func (o *Oex) GetUserInfo() (UserInfoResponse, error) {
+	var resp UserInfoResponse
+	path := o.APIUrl + oexUserInfo
+	err := o.SendAuthHTTPRequest(http.MethodGet, path, nil, &resp)
+	if err != nil {
+		return resp, err
+	}
+
+	if resp.Error != oexNoError {
+		return resp, errors.New(resp.ErrCapture.Msg)
+	}
+
+	return resp, nil
+}
+
+// GetAllOrders acquires full delegation
+func (o *Oex) GetAllOrders(symbol, startDate, endDate, pageSize, page string) (AllOrderResponse, error) {
+	var resp AllOrderResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	params.Set("startDate", startDate)
+	params.Set("endDate", endDate)
+	params.Set("pageSize", pageSize)
+	params.Set("page", page)
+	path := o.APIUrl + oexAllOpenOrders
+	err := o.SendAuthHTTPRequest(http.MethodGet, path, params, &resp)
+	if err != nil {
+		return resp, err
+	}
+
+	if resp.Error != oexNoError {
+		return resp, errors.New(resp.ErrCapture.Msg)
+	}
+
+	return resp, nil
+}
+
+// FindOrderHistory fetches past orders based on the parameters
+func (o *Oex) FindOrderHistory(symbol, startDate, endDate, pageSize, page, sort string) (OrderHistoryResponse, error) {
+	var resp OrderHistoryResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	params.Set("startDate", startDate)
+	params.Set("endDate", endDate)
+	params.Set("pageSize", pageSize)
+	params.Set("page", page)
+	params.Set("sort", sort)
+	path := o.APIUrl + oexOrderHistory
+	err := o.SendAuthHTTPRequest(http.MethodGet, path, params, &resp)
+	if err != nil {
+		return resp, err
+	}
+
+	if resp.Error != oexNoError {
+		return resp, errors.New(resp.ErrCapture.Msg)
+	}
+
+	return resp, nil
+}
+
+// RemoveOrder cacncels the order for the provided OrderID
+func (o *Oex) RemoveOrder(orderID, symbol string) (RemoveOrderResponse, error) {
+	var resp RemoveOrderResponse
+	params := url.Values{}
+	params.Set("order_id", orderID)
+	params.Set("symbol", symbol)
+	path := o.APIUrl + oexRemoveOrder
+	err := o.SendAuthHTTPRequest(http.MethodPost, path, params, &resp)
+	if err != nil {
+		return resp, err
+	}
+
+	if resp.Error != oexNoError {
+		return resp, errors.New(resp.ErrCapture.Msg)
+	}
+
+	return resp, nil
+}
+
+// RemoveAllOrders cancels all orders for a given currency pair
+func (o *Oex) RemoveAllOrders(symbol string) (RemoveOrderResponse, error) {
+	var resp RemoveOrderResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	path := o.APIUrl + oexRemoveAllOrder
+	err := o.SendAuthHTTPRequest(http.MethodPost, path, params, &resp)
+	if err != nil {
+		return resp, err
+	}
+
+	if resp.Error != oexNoError {
+		return resp, errors.New(resp.ErrCapture.Msg)
+	}
+
+	return resp, nil
+
+}
+
+// CreateOrder creates a new order
+func (o *Oex) CreateOrder(side, orderType, volume, price, symbol, fee string) (CreateOrderResponse, error) {
```

Comment:
Please adjust the params to more preferred types (volume/price being float64s). Some other methods require the same treatment.